### PR TITLE
fix(table): table checkbox not disabling when no data presents

### LINKED ISSUE #3230: Unable to disable `Table` checkbox.
### Which package has an issue?

@yamada-ui/react

### Package Version

1.5.5

### Description

Unable to disable `Table` checkbox.

In this screen, the checkbox is disabled when there were no records.

Screenshot 2024-10-24 at 19 25 31

### Link to Reproduction

https://yamada-ui.com/examples/tasks

### Steps to Reproduce

2. Click on '...'
3. Scroll down to '...'

### Browser

None

### Operating System

- [ ] macOS
- [ ] Windows
- [ ] Linux
- [ ] iPadOS
- [ ] iOS
- [ ] Android

### Do you know how to fix the issue?

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix?

None

### Additional Information

Probably due to the `rest` being overwritten.

```tsx
<Checkbox
  ref={isDisabled ? undefined : ref}
  tabIndex={tabIndex}
  inputProps={{ "aria-label": "Select all row" }}
  {...{ gap: 0, ...rest }}
  isChecked={isAllChecked || isChecked}
  isDisabled={isDisabled}
  {...(!isAllChecked ? { isIndeterminate } : {})}
  onChange={handlerAll(rest.onChange, onChange)}
/>
```

### PR DESCRIPTION
**Pull request by nathanlao**
Closes #3230

## Description

The `{...rest}` spread is before `isDisabled` prop,  it will be overwritten by the `isDisabled` value, the checkbox behavior is affected.

## Current behavior (updates)

 Table checkbox is not disabling when it contains no data after filtering

## New behavior

 Table checkbox is disabling when it contains no data after filtering

image

## Is this a breaking change (Yes/No):

No

## Additional Information

Feel free to let me know if there is any better fix that is good for code maintenance

### CODE REVIEW

**Review by koralle — DISMISSED** [review]

File: packages/components/table/src/table-checkbox.tsx [hunk 1]
```diff
@@ -75,13 +75,14 @@ export const TableTotalCheckbox = forwardRef(
...
-          tabIndex={tabIndex}
-          inputProps={{ "aria-label": "Select all row" }}
-          {...{ gap: 0, ...rest }}
+          gap={0}
           isChecked={isAllChecked || isChecked}
           isDisabled={isDisabled}
+          tabIndex={tabIndex}
+          inputProps={{ "aria-label": "Select all row" }}
           {...(!isAllChecked ? { isIndeterminate } : {})}
           onChange={handlerAll(rest.onChange, onChange)}
```
Comment (hirotomoyamada) [inline packages/components/table/src/table-checkbox.tsx:84]: This line must be after `{..rest}` and must be later than `{...rest}`.

Comment (hirotomoyamada) [reply packages/components/table/src/table-checkbox.tsx:84]: Please sort them like this.

```tsx
<Checkbox
  ref={isDisabled ? undefined : ref}
  isChecked={isAllChecked || isChecked}
  isDisabled={isDisabled}
  tabIndex={tabIndex}
  inputProps={{ "aria-label": "Select all row" }}
  {...(!isAllChecked ? { isIndeterminate } : {})}
  {...{ gap: 0, ...rest }}
  onChange={handlerAll(rest.onChange, onChange)}
/>
```

Comment (nathanlao) [reply packages/components/table/src/table-checkbox.tsx:84]: I just made the change, so the following props should be before `{...rest}` to ensure they can be overwritten by `{...rest}` props?

```js
isChecked={isAllChecked || isChecked}
isDisabled={isDisabled}
tabIndex={tabIndex}
inputProps={{ "aria-label": "Select all row" }}
```

Comment (hirotomoyamada) [reply packages/components/table/src/table-checkbox.tsx:84]: Yes, Yamada UI must be provided so that developers can override everything.

**Review by hirotomoyamada — DISMISSED** [review]

**Review by hirotomoyamada — CHANGES_REQUESTED** [review]
Please run `pnpm changeset` and select `patch`.

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**